### DISCUSSION TITLE
能否实现画中画小窗？缩小到右上角的那种

### DISCUSSION THREAD

**Discussion by zzyyi66** [hooke007/mpv_PlayKit#206] [mpv 通用]
类似mac的IINA播放器

**Comment by natural-harmonia-gropius** [comment]
不能，原版mpv的窗口是能用就行的那种

**Comment by zhongfly** [comment]
你自己把窗口**置顶**缩小不就是了……

**Reply by hooke007** [reply]
他想的应该是通过系统PIP api实现的画中画，有一些独特的窗口特性（例如win10的pip窗口不会被win+d最小化）。

**Reply by zzyyi66** [reply]
自己缩小只能缩小到中间，拖到右上角放大后，位置就变了，画面只有一部分在屏幕外面

**Reply by hooke007** [reply]
<s>如果只是锁死窗口尺寸，这个脚本可以
https://github.com/hooke007/MPV_lazy/blob/img/portable_config/scripts/freeze_window.lua</s>

（当然受限于windows的实现不完全 https://github.com/mpv-player/mpv/issues/8379 ，导入不同分辨率的视频的时候还是会变更窗口大小

**Comment by nyfair** [comment]
换个思路，缩小到右上角之后整成桌面动态壁纸嵌入到windows背景里，类似小范围的wallpaper engine能满足你需求麽

**Comment by hooke007** [comment]
> 自己缩小只能缩小到中间，拖到右上角放大后，位置就变了，画面只有一部分在屏幕外面

伴随近期更新的参数 `--auto-window-resize=no` 的加入，围绕它可以做一个伪画中画，可满足当前主题的基础需求。
使用主线分支的脚本 [input_plus.lua](https://github.com/hooke007/MPV_lazy/blob/main/portable_config/scripts/input_plus.lua) 的预设命令 `script-binding pip_dummy` ；或者你可以用快捷键+profile联合制定自己的“画中画”方案。

**Comment by verygoodlee** [comment]
我实现了一个 **真·画中画（99真）**[ mpv-pip](https://github.com/verygoodlee/mpv-pip)，窗口大小调整，窗口对齐到角落，置顶且无法最小化，这些功能都实现了，体验上和原生画中画很接近了。
最小化时窗口有瞬间闪烁，因为实现方式是在最小化的瞬间把窗口重新拉起来。
